### PR TITLE
Add Mongo to SQL connector list [HZ-2502]

### PR DESCRIPTION
**Pull request by TomaszGaweda**

### CODE REVIEW

File: docs/modules/sql/pages/mapping-to-mongo.adoc [hunk 1]
```diff
@@ -165,6 +163,12 @@ DATA CONNECTION myMongo
 ----
 Notice that there is no mention of `TYPE MONGO` this time; it’s automatically assumed by the SQL engine when you provide MongoDB data connection. This works with both schema provided or not.
 
+*Note*: You have to specify database name. You have 3 options:
+
+ - add `OPTIONS ('database' = 'myDatabase')` in CREATE DATA CONNECTION
+ - add `OPTIONS ('database' = 'myDatabase')` in CREATE MAPPING
+ - use full external name, e.g. `CREATE MAPPING people EXTERNAL NAME myDatabase.people (...)`
```
Comment (Serdaro) [inline docs/modules/sql/pages/mapping-to-mongo.adoc:170]:
```suggestion

[NOTE]
====
You have to specify the database name using one of the following options:

* add `OPTIONS ('database' = 'myDatabase')` in `CREATE DATA CONNECTION`
* add `OPTIONS ('database' = 'myDatabase')` in `CREATE MAPPING`
* use the full external name, e.g., `CREATE MAPPING people EXTERNAL NAME myDatabase.people (...)`
====
```

**Review by Serdaro — APPROVED** [review]
Minor syntax suggestion for a note.

File: docs/modules/sql/pages/mapping-to-mongo.adoc [hunk 2]
```diff
@@ -165,6 +163,73 @@ DATA CONNECTION myMongo
...
+----
+--
+====
+
+== Field mappings
+
+Object type `Collection` reads columns to be the same as properties in Mongo collection.
+
+For `ChangeStream` it's more complicated: the ChangeStream objects have Mongo collection's properties prefixed with `fullDocument.`,
+There are also some predefined, top-level (without `fullDocument.` prefix) columns
```
Comment (angelasimms) [inline docs/modules/sql/pages/mapping-to-mongo.adoc:224]:
```suggestion
There are also some predefined, top-level columns without the `fullDocument.` prefix:
```

File: docs/modules/sql/pages/mapping-to-mongo.adoc [hunk 3]
```diff
@@ -165,6 +163,73 @@ DATA CONNECTION myMongo
...
+
+ - `operationType STRING` - operation that triggered this Change event, e.g. `insert`
```
Comment (angelasimms) [inline docs/modules/sql/pages/mapping-to-mongo.adoc:226]:
```suggestion
- `operationType STRING`: Operation that triggered this change event, e.g. `insert`
```

File: docs/modules/sql/pages/mapping-to-mongo.adoc [hunk 4]
```diff
@@ -165,6 +163,73 @@ DATA CONNECTION myMongo
...
+- `resumeToken STRING` - resume token associated with given change stream event.
```
Comment (angelasimms) [inline docs/modules/sql/pages/mapping-to-mongo.adoc:227]:
```suggestion
- `resumeToken STRING`: Resume token associated with the given change stream event.
```

File: docs/modules/sql/pages/mapping-to-mongo.adoc [hunk 5]
```diff
@@ -165,6 +163,73 @@ DATA CONNECTION myMongo
...
+- `wallTime DATE_TIME` - wall time of the event.
```
Comment (angelasimms) [inline docs/modules/sql/pages/mapping-to-mongo.adoc:228]: Not sure this is very useful if you don't know what "wall time" is. Could you give a fuller description?

File: docs/modules/sql/pages/mapping-to-mongo.adoc [hunk 6]
```diff
@@ -165,6 +163,73 @@ DATA CONNECTION myMongo
...
+- `ts TIMESTAMP` - Timestamp of when the even occurred, either equal to wall time or to current time on the Hazelcast node,
```
Comment (angelasimms) [inline docs/modules/sql/pages/mapping-to-mongo.adoc:229]:
```suggestion
- `ts TIMESTAMP`: Timestamp when the event occurred, which is either equal to wall time on the Hazelcast member, or to current time if no wall time was provided. 
```

File: docs/modules/sql/pages/mapping-to-mongo.adoc [hunk 7]
```diff
@@ -165,6 +163,73 @@ DATA CONNECTION myMongo
...
+  if no wall time was provided.
+- `clusterTime TIMESTAMP` - cluster time on which this even occurred.
+
 == Type Mapping
 
 The type system in MongoDB and SQL is not exactly the same. That leads to potential confusions and the need of type coercion.
```
Comment (angelasimms) [inline docs/modules/sql/pages/mapping-to-mongo.adoc:235]:
```suggestion
The type system in MongoDB and SQL is not exactly the same. This leads to potential confusion and the need for the type coercion.
```

**Review by angelasimms — CHANGES_REQUESTED** [review]
Added some suggestions and questions. Please check carefully as I don't want to change the meaning of anything you have written.

File: docs/modules/sql/pages/mapping-to-mongo.adoc [hunk 8]
```diff
@@ -165,6 +163,73 @@ DATA CONNECTION myMongo
...
+OPTIONS (...)
+----
+--
+====
+
+== Field mappings
+
+Object type `Collection` reads columns to be the same as properties in Mongo collection.
+
+For `ChangeStream` it's more complicated: the ChangeStream objects have Mongo collection's properties prefixed with `fullDocument.`,
```
[the remark this answers is not in the document]
Comment (TomaszGaweda) [reply docs/modules/sql/pages/mapping-to-mongo.adoc:223]:
```suggestion
For `ChangeStream` it is more complicated. The ChangeStream objects contains the MongoDB collection properties, prefixed with `fullDocument.`.
```

I don't really "like" `take` here - changeStream object contains original event data, I think `contains` is better here, WDYT?

[the remark this answers is not in the document]
Comment (angelasimms) [reply docs/modules/sql/pages/mapping-to-mongo.adoc:223]: I agree, that makes sense.

File: docs/modules/sql/pages/mapping-to-mongo.adoc [hunk 9]
```diff
@@ -218,17 +218,19 @@ OPTIONS (...)
 
 == Field Mappings
 
-Object type `Collection` reads columns to be the same as properties in Mongo collection.
+Object type `Collection` resolve columns to be the same as Mongo collection properties' names and types.
```
Comment (angelasimms) [inline docs/modules/sql/pages/mapping-to-mongo.adoc:222]: Would it be correct to say this?

```suggestion
Object type `Collection` resolves columns to the names and types properties of a Mongo collection.
```

File: docs/modules/sql/pages/mapping-to-mongo.adoc [hunk 10]
```diff
@@ -218,17 +218,19 @@ OPTIONS (...)
...
 For `ChangeStream` it's more complicated: the ChangeStream objects have Mongo collection's properties prefixed with `fullDocument.`,
 There are also some predefined, top-level columns without the `fullDocument.` prefix:
 
 - `operationType STRING`: Operation that triggered this change event, e.g. `insert`
 - `resumeToken STRING`: Resume token associated with the given change stream event.
-- `wallTime DATE_TIME` - wall time of the event.
-- `ts TIMESTAMP` - Timestamp of when the even occurred, either equal to wall time or to current time on the Hazelcast node,
-  if no wall time was provided.
-- `clusterTime TIMESTAMP` - cluster time on which this even occurred.
+- `wallTime DATE_TIME` - wall time of the event, which means server's date and time of the database operation.
```
Comment (angelasimms) [inline docs/modules/sql/pages/mapping-to-mongo.adoc:228]: Does this work?
```suggestion
- `wallTime DATE_TIME`: Wall time of the event. The date and time of the database operation on the server.
```

File: docs/modules/sql/pages/mapping-to-mongo.adoc [hunk 11]
```diff
@@ -218,17 +218,19 @@ OPTIONS (...)
...
+- `ts TIMESTAMP` - Timestamp of when the even occurred, either equal to wall time of the even (see above) or to current
```
Comment (angelasimms) [inline docs/modules/sql/pages/mapping-to-mongo.adoc:229]: Is this accurate?

```suggestion
- `ts TIMESTAMP`: Timestamp of the event. This is either equal to wall time of the event, if provided, or to current time on the Hazelcast member. 
```

File: docs/modules/sql/pages/mapping-to-mongo.adoc [hunk 12]
```diff
@@ -218,17 +218,19 @@ OPTIONS (...)
...
+  time on the Hazelcast node, if no event's wall time was provided.
```
Comment (angelasimms) [inline docs/modules/sql/pages/mapping-to-mongo.adoc:230]:
```suggestion
```

File: docs/modules/sql/pages/mapping-to-mongo.adoc [hunk 13]
```diff
@@ -218,17 +218,19 @@ OPTIONS (...)
...
+- `clusterTime TIMESTAMP` - cluster time on which this even occurred, which means timestamp associated with Mongo oplog entry.
```
Comment (angelasimms) [inline docs/modules/sql/pages/mapping-to-mongo.adoc:231]:
```suggestion
- `clusterTime TIMESTAMP`: Cluster time of the event. The timestamp associated with Mongo oplog entry.
```

File: docs/modules/sql/pages/mapping-to-mongo.adoc [hunk 14]
```diff
@@ -218,17 +218,19 @@ OPTIONS (...)
...
+
+Recommended Mongo documentation page: https://www.mongodb.com/docs/manual/reference/change-events/create/
```
Comment (angelasimms) [inline docs/modules/sql/pages/mapping-to-mongo.adoc:233]:
```suggestion
For further information, see the link:https://www.mongodb.com/docs/manual/reference/change-events/create/[Mongo documentation].  
```

**Review by angelasimms — CHANGES_REQUESTED** [review]
Sorry, just a few more minor comments.

File: docs/modules/sql/pages/mapping-to-mongo.adoc [hunk 15]
```diff
@@ -165,9 +164,77 @@ DATA CONNECTION myMongo
...
+== Supported Object Types
+
+There may be one or more object types for a connector. The Mongo SQL connector has two object types:
+
+ - `Collection`: Represents a batch read from a given collection. This is the default object type.
+ - `ChangeStream`: Represents reading a stream of events.
+
+To change the object type, append `OBJECT TYPE X` after `DATA CONNECTION` / `TYPE`. For example:
+
+[tabs]
```
Comment (angelasimms) [inline docs/modules/sql/pages/mapping-to-mongo.adoc:186]: The tabs are not displaying in the preview as the syntax is slightly wrong. Try this instead:

```[tabs]
====
Using Data Connection::
+
--
[source,sql]
----
CREATE MAPPING people (
    firstName VARCHAR(100),
    lastName VARCHAR(100),
    age INT
)
DATA CONNECTION myMongo
OBJECT TYPE ChangeStream
OPTIONS (...)
----
--
Using TYPE::
+
--
[source,sql]
----
CREATE MAPPING people (
firstName VARCHAR(100),
lastName VARCHAR(100),
age INT
)
TYPE Mongo
OBJECT TYPE ChangeStream
OPTIONS (...)
----
--
====

File: docs/modules/sql/pages/mapping-to-mongo.adoc [hunk 16]
```diff
@@ -165,9 +164,77 @@ DATA CONNECTION myMongo
...
+TYPE Mongo
+OBJECT TYPE ChangeStream
+OPTIONS (...)
+----
+--
+====
+
+== Field Mappings
+
+Object type `Collection` resolve columns to be the same as Mongo collection properties' names and types.
```
Comment (angelasimms) [inline docs/modules/sql/pages/mapping-to-mongo.adoc:222]:
```suggestion

```
This is a duplicate of the sentence below.

File: docs/modules/sql/pages/mapping-to-mongo.adoc [hunk 17]
```diff
@@ -165,9 +164,77 @@ DATA CONNECTION myMongo
...
+Object type `Collection` resolves columns to the names and types properties of a Mongo collection.
+For `ChangeStream` it is more complicated. The ChangeStream objects contains the MongoDB collection properties, prefixed with `fullDocument.`.
```
Comment (angelasimms) [inline docs/modules/sql/pages/mapping-to-mongo.adoc:224]:
```suggestion
For `ChangeStream` it is more complicated. The `ChangeStream` object contains the MongoDB collection properties, prefixed with `fullDocument.`.
```

File: docs/modules/sql/pages/mapping-to-mongo.adoc [hunk 18]
```diff
@@ -165,9 +164,77 @@ DATA CONNECTION myMongo
...
+There are also some predefined, top-level columns without the `fullDocument.` prefix:
+
+- `operationType STRING`: Operation that triggered this change event, e.g. `insert`
```
Comment (angelasimms) [inline docs/modules/sql/pages/mapping-to-mongo.adoc:227]:
```suggestion
- `operationType STRING`: Operation that triggered this change event, e.g. `insert`.
```

File: docs/modules/sql/pages/mapping-to-mongo.adoc [hunk 19]
```diff
@@ -165,9 +164,77 @@ DATA CONNECTION myMongo
...
+- `resumeToken STRING`: Resume token associated with the given change stream event.
+- `wallTime DATE_TIME`: Wall time of the event. The date and time of the database operation on the server.
+- `ts TIMESTAMP`: Timestamp of the event. This is either equal to wall time of the event, if provided, or to current time on the Hazelcast member.
```
Comment (angelasimms) [inline docs/modules/sql/pages/mapping-to-mongo.adoc:230]:
```suggestion
- `ts TIMESTAMP`: Timestamp of the event. This is either equal to the wall time of the event, if provided, or to the current time on the Hazelcast member. 
```

**Review by angelasimms — CHANGES_REQUESTED** [review]
Thanks for making the other changes. These should be the last ones.

File: docs/modules/sql/pages/mapping-to-mongo.adoc [hunk 20]
```diff
@@ -165,9 +164,77 @@ DATA CONNECTION myMongo
...
+OBJECT TYPE ChangeStream
+OPTIONS (...)
+----
+--
+====
+
+== Field Mappings
+
+Object type `Collection` resolve columns to be the same as Mongo collection properties' names and types.
+Object type `Collection` resolves columns to the names and types properties of a Mongo collection.
```
Comment (TomaszGaweda) [inline docs/modules/sql/pages/mapping-to-mongo.adoc:223]:
```suggestion
Object type `Collection` resolves columns to the names and types of the properties of a Mongo collection.
```

File: docs/modules/sql/pages/mapping-to-mongo.adoc (hunk 15 above)
Comment (TomaszGaweda) [reply docs/modules/sql/pages/mapping-to-mongo.adoc:186]: Done: https://github.com/hazelcast/hz-docs/pull/761/commits/36b3c1aeb14ce3e00923f86ca9f95dff8ac1f68a

**Review by angelasimms — APPROVED** [review]
LGTM. Thank you for your patience with this.

**Review by frant-hartm — APPROVED** [review]

**Pull request merged**